src/medali/core.py

```diff
@@ -0,0 +1,377 @@
+                        self._meta[key] = str(value)
+                    else:
+                        err_msg = "Metadata information for key '{}' has to be boolean.".format(key)
+                        raise ValueError(err_msg)
+                elif self._ref_meta['Metadata'][key] == 'integer':
+                    if isinstance(value, int):
+                        self._meta[key] = str(value)
+                    else:
+                        err_msg = "Metadata information for key '{}' has to be integer".format(key)
+                        raise ValueError(err_msg)
+                elif self._ref_meta['Metadata'][key] == 'number':
+                    if isinstance(value, numbers.Number):
+                        self._meta[key] = str(value)
+                    else:
+                        err_msg = "Metadata information for key '{}' has to be number".format(key)
+                        raise ValueError(err_msg)
+                elif self._ref_meta['Metadata'][key] == 'datetime':
+                    if isinstance(value, datetime.datetime):
+                        self._meta[key] = value.strftime("%Y-%m-%d %H:%M:%S")
+                    else:
+                        err_msg = "Metadata information for key '{}' has to be datetime object".format(key)
+                        raise ValueError(err_msg)
+                else:
+                    self._meta[key] = str(value)
+            else:
+                err_msg = "Metadata attribute '{}' is not given in the product reference.".format(key)
+                raise KeyError(err_msg)
+        else:
+            self._meta[key] = str(value)
+
+    def _get_metadata(self, attribute):
+        """
+        Decodes and returns metadata value according to the given attribute.
+
+        Parameters
+        ----------
+        attribute : str
+            Metadata attribute.
+
+        Returns
+        -------
+        value : any
+            Decoded metadata value.
+
+        """
+        if attribute not in self._meta.keys():
+            err_msg = "Metadata attribute '{}' can not be found.".format(attribute)
+            raise KeyError(err_msg)
+
+        if self._ref_meta:
+            if self._meta[attribute] == 'null':
+                value = None
+            elif self._ref_meta['Metadata'][attribute] == 'boolean':
+                value = self._meta[attribute] == 'True'
+            elif self._ref_meta['Metadata'][attribute] == 'integer':
+                value = int(self._meta[attribute])
+            elif self._ref_meta['Metadata'][attribute] == 'number':
+                value = float(self._meta[attribute])
+            elif self._ref_meta['Metadata'][attribute] == 'datetime':
+                value = datetime.datetime.strptime(self._meta[attribute], "%Y-%m-%d %H:%M:%S")
+            else:
+                value = self._meta[attribute]
+        else:
+            value = self._meta[attribute]
+
+        return value
+
+    def check_metadata(self, metadata):
```

Comment:
While initializing the MetaData class there is already a metadata dictionary read in which contains decoded values, while in this function a metadata dictionary with encoded values is needed. Could we use the already existing dictionary here instead?

Reply:
Yes, but then we need more effort on decoding and encoding the attributes and would need to change the class interaction with metadata coming from a file, in string format. I will try to change this, that the class becomes more compact.

Reply:
In my opinion it is not a big deal to leave it like this. I just wanted to know if it is maybe only a mistake which could be fixed easily.